### PR TITLE
Divide module and support multi module project

### PR DESCRIPTION
**Pull request by ksmail13**
resolved: #11 
- Divide modules by related application

---

- Support multi-module project (multi coverage reports)

### CODE REVIEW

**Comment by ksmail13** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 78 / 94 (82.98%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/naver/nid/cover/parser/coverage/CoverageReportXmlHandler.java|0|1|00.00%|
|:large_blue_circle:|com/naver/nid/cover/github/manager/GithubPullRequestManager.java|0|5|00.00%|
|:large_blue_circle:|com/naver/nid/cover/util/ObjectFactory.java|3|8|37.50%|
|:large_blue_circle:|com/naver/nid/cover/parser/diff/FileDiffReader.java|2|3|66.67%|
|:large_blue_circle:|com/naver/nid/cover/CoverChecker.java|3|4|75.00%|
|:large_blue_circle:|com/naver/nid/cover/github/reporter/GithubPullRequestReporter.java|56|59|94.92%|
|:large_blue_circle:|com/naver/nid/cover/checker/NewCoverageChecker.java|3|3|100.00%|
|:large_blue_circle:|com/naver/nid/cover/cobertura/CoberturaCoverageReportHandler.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/github/parser/GithubDiffReader.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/jacoco/JacocoReportParser.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/parser/diff/DiffParser.java|2|2|100.00%|
|:large_blue_circle:|com/naver/nid/cover/jacoco/JacocoXmlCoverageReportHandler.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/util/ParameterParser.java|5|5|100.00%|

File: cover-checker-console/src/main/resources/logback.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <logger name="com.naver.nid.cover.parser.diff" level="INFO" />
     <logger name="com.naver.nid.cover.parser.coverage" level="INFO" />
     <logger name="com.naver.nid.cover.checker" level="INFO" />
-    <logger name="com.naver.nid.cover.github" level="INFO" />
+    <logger name="com.naver.nid.cover.com.naver.nid.cover.github" level="INFO" />
```
Comment (moonsoo5522) [inline cover-checker-console/src/main/resources/logback.xml]: 오타인가요?

File: cover-checker-github/src/main/java/com/naver/nid/cover/github/manager/GithubPullRequestManager.java [hunk 2]
```diff
@@ -46,22 +46,22 @@ public GithubPullRequestManager(GitHubClient github, String repoName, int prNumb
...
-		return new GithubCommentManager(new IssueService(ghClient), repoId, pr.getNumber());
+		return new GithubCommentManager(new IssueService(ghClient), repoId, prNumber);
 	}
 
 	public GithubStatusManager statusManager() {
+		try {
+			pr = prService.getPullRequest(repoId, this.prNumber);
+		} catch (IOException e) {
+			throw new IllegalStateException(e);
+		}
```
Comment (moonsoo5522) [inline cover-checker-github/src/main/java/com/naver/nid/cover/github/manager/GithubPullRequestManager.java]: 이 코드를 생성자에서 메서드로 옮기신 이유가 있나요?

커버체커의 생명주기가 한 PR내에서 모두 끝날 것이라, 생성자에 그대로 있는게 자연스러워 보여서요~
(이 메서드가 반드시 불리지는 않는 것인지?)

File: cover-checker-jacoco/src/test/resources/reports/html/com.naver.nid.cover.github/GithubCommentManager.java.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../jacoco-resources/report.gif" type="image/gif"/><title>GithubCommentManager.java</title><link rel="stylesheet" href="../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">coverchecker</a> &gt; <a href="index.source.html" class="el_package">com.naver.nid.cover.com.naver.nid.cover.github</a> &gt; <span class="el_source">GithubCommentManager.java</span></div><h1>GithubCommentManager.java</h1><pre class="source lang-java linenums">package com.naver.nid.cover.com.naver.nid.cover.github;
```
Comment (moonsoo5522) [inline cover-checker-jacoco/src/test/resources/reports/html/com.naver.nid.cover.github/GithubCommentManager.java.html]: 이건 테스트용 파일인가요?

File: cover-checker-console/src/main/resources/logback.xml (hunk 1 above)
Comment (ksmail13) [reply cover-checker-console/src/main/resources/logback.xml]: 👍

File: cover-checker-github/src/main/java/com/naver/nid/cover/github/manager/GithubPullRequestManager.java (hunk 2 above)
Comment (ksmail13) [reply cover-checker-github/src/main/java/com/naver/nid/cover/github/manager/GithubPullRequestManager.java]: 처음에 의존성 이슈로 저렇게 수정했었던것 같은데,
다시보니 원래 방식이 이슈는 없을 것 같습니다.

File: cover-checker-jacoco/src/test/resources/reports/html/com.naver.nid.cover.github/GithubCommentManager.java.html (hunk 3 above)
Comment (ksmail13) [reply cover-checker-jacoco/src/test/resources/reports/html/com.naver.nid.cover.github/GithubCommentManager.java.html]: 네 맞습니다.

**Comment by ksmail13** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 79 / 92 (85.87%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/naver/nid/cover/parser/coverage/CoverageReportXmlHandler.java|0|1|00.00%|
|:large_blue_circle:|com/naver/nid/cover/github/manager/GithubPullRequestManager.java|0|1|00.00%|
|:large_blue_circle:|com/naver/nid/cover/util/ObjectFactory.java|3|8|37.50%|
|:large_blue_circle:|com/naver/nid/cover/CoverChecker.java|3|5|60.00%|
|:large_blue_circle:|com/naver/nid/cover/parser/diff/FileDiffReader.java|2|3|66.67%|
|:large_blue_circle:|com/naver/nid/cover/github/reporter/GithubPullRequestReporter.java|56|59|94.92%|
|:large_blue_circle:|com/naver/nid/cover/checker/NewCoverageChecker.java|3|3|100.00%|
|:large_blue_circle:|com/naver/nid/cover/cobertura/CoberturaCoverageReportHandler.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/github/parser/GithubDiffReader.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/jacoco/JacocoReportParser.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/parser/diff/DiffParser.java|2|2|100.00%|
|:large_blue_circle:|com/naver/nid/cover/jacoco/JacocoXmlCoverageReportHandler.java|1|1|100.00%|
|:large_blue_circle:|com/naver/nid/cover/util/ParameterParser.java|6|6|100.00%|

**Pull request merged**